### PR TITLE
Add AsProxy property to AutoMapAttribute

### PR DESCRIPTION
**Pull request by korser1**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AutoMapper/AutoMapper?pullRequest=3839) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AutoMapper/AutoMapper?pullRequest=3839) before we can accept your contribution.<br/><br/>:x: korser1 [sign now](https://cla.dotnetfoundation.org/AutoMapper/AutoMapper?pullRequest=3839)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AutoMapper/AutoMapper?pullRequest=3839) it.</sub>

File: src/AutoMapper/Configuration/Annotations/AutoMapAttribute.cs [hunk 1]
```diff
@@ -81,6 +86,11 @@ public void ApplyConfiguration(IMappingExpression mappingExpression)
             {
                 mappingExpression.ConvertUsing(TypeConverter);
             }
+
+            if (AsProxy)
```
Comment (korser1) [inline src/AutoMapper/Configuration/Annotations/AutoMapAttribute.cs:90]: Here could be a check if Dest type is interface or not

Comment (lbargaoanu) [reply src/AutoMapper/Configuration/Annotations/AutoMapAttribute.cs:90]: That would be the job of the fluent API. So no need.

File: src/UnitTests/AttributeBasedMaps.cs [hunk 2]
```diff
@@ -953,5 +953,58 @@ public void Should_validate_successfully()
...
+            protected override MapperConfiguration CreateConfiguration() => new(cfg =>
+            {
+                cfg.AddMaps(typeof(When_not_specifying_as_proxy_via_attribute));
+            });
+
+            [Fact]
+            public void Should_not_convert_to_interface()
+            {
+                var source = new Source { Value = 15 };
+                typeof(AutoMapperMappingException).ShouldBeThrownBy(() => Mapper.Map<IDest>(source));
```
Comment (lbargaoanu) [inline src/UnitTests/AttributeBasedMaps.cs:1006]: If we have this  test, we should check smth about the exception, so we know why it failed.

**Review by lbargaoanu — APPROVED** [review]

File: src/UnitTests/AttributeBasedMaps.cs (hunk 2 above)
Comment (korser1) [reply src/UnitTests/AttributeBasedMaps.cs]: Please review the updates

**Pull request merged**